Client/Frontend/Settings/NotificationsSettingsViewController.swift

```diff
@@ -0,0 +1,179 @@
+                NotificationCenter.default.post(name: .RegisterForPushNotifications, object: nil)
+            }
+        }
+    }()
+
+    private lazy var tipsAndFeaturesNotifications: BoolNotificationSetting = {
+        return BoolNotificationSetting(
+            title: .Settings.Notifications.TipsAndFeaturesNotificationsTitle,
+            description: String(format: .Settings.Notifications.TipsAndFeaturesNotificationsStatus, AppName.shortName.rawValue),
+            prefs: prefs,
+            prefKey: PrefsKeys.Notifications.TipsAndFeaturesNotifications,
+            enabled: true
+        ) { [weak self] value in
+            guard let self = self else { return }
+
+            Task {
+                let shouldEnable = await self.notificationsChanged(value)
+                self.tipsAndFeaturesNotifications.control.setOn(shouldEnable, animated: true)
+                self.tipsAndFeaturesNotifications.writeBool(self.tipsAndFeaturesNotifications.control)
+
+                if shouldEnable {
+                    // schedule engagement notifications if necessary
+                    self.engagementNotificationHelper.schedule()
+                } else {
+                    // cancel all pending engagement notifications
+                    self.engagementNotificationHelper.cancelAll()
+                }
+            }
+        }
+    }()
+
+    private let prefs: Prefs
+    private let hasAccount: Bool
+    private var footerTitle = ""
+
+    init(prefs: Prefs, hasAccount: Bool) {
+        self.prefs = prefs
+        self.hasAccount = hasAccount
+        super.init(style: .grouped)
+        self.title = .Settings.Notifications.Title
+        self.addObservers()
+
+        Task {
+            await self.checkForSystemNotifications()
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func generateSettings() -> [SettingSection] {
+        let childrenSection: [Setting]
+        if hasAccount {
+            childrenSection = [syncNotifications, tipsAndFeaturesNotifications]
+        } else {
+            childrenSection = [tipsAndFeaturesNotifications]
+        }
+
+        return [
+            SettingSection(footerTitle: NSAttributedString(string: footerTitle), children: childrenSection)
+        ]
+    }
+
+    func checkForSystemNotifications() async {
+        let settings = await NotificationManager().getNotificationSettings()
+
+        switch settings.authorizationStatus {
+        case .authorized, .provisional, .ephemeral:
+            self.footerTitle = ""
+        case .denied:
+            self.footerTitle = String(format: .Settings.Notifications.systemNotificationsDisabledMessage, AppName.shortName.rawValue, AppName.shortName.rawValue)
+        case .notDetermined:
+            self.footerTitle = ""
+        @unknown default:
+            self.footerTitle = ""
+        }
```

Comment:
Would this be a cleaner implementation of this switch? So you don't have to write `self.footerTitle = ""` a bunch of times. I may be misunderstanding the reasoning behind why it was written like this, though, so, feel free to correct me.

```suggestion
        switch settings.authorizationStatus {
        case .denied:
            self.footerTitle = String(format: .Settings.Notifications.systemNotificationsDisabledMessage, AppName.shortName.rawValue, AppName.shortName.rawValue)
        case .authorized, .provisional, .ephemera, .notDetermined:
            fallthrough
        @unknown default:
            self.footerTitle = ""
        }
```

The other alternative, is just having the `.denied` and a `default` case, seeing as you're really only doing one of two things?